src/index.ts

```diff
@@ -1,100 +1,79 @@
+.use(cors())
+.use(databasePlugin)
 
-  newObject.id = newId
-
-  // Add the new object to the data array
-  data.push(newObject)
+app.state('version',1)
+.get("/all", ({ dbAction, database })=> {
+  const result = dbAction({
+    database: database(),
+    queryMethod: 'all',
+    query: "SELECT * FROM todos",
+    errorMessage: "Could not get all todo items",
+  })
 
-  // Return the modified data
-  return { data }
+  return result
 })
-.decorate('editTodoById', (id: number, updatedData: { title?: string; description?: string; isComplete?: boolean }) => {
-  // Find the index of the object with the specified ID
-  const todoIndex = data.findIndex(item => item.id === id);
 
-  if (todoIndex === -1) {
-    throw new Error("Could not find todo item")
-  }
-  // Perform the edit based on the provided properties in updatedData
-  data[todoIndex] = {
-    ...data[todoIndex],
-    ...updatedData
-  }
+.get('/todo/:id', ({params: {id}, dbAction, database})=>{
+  const result = dbAction({
+    database: database(),
+    queryMethod: 'get',
+    query: "SELECT * FROM todos WHERE id = ?",
+    params: [id],
+    errorMessage: "Could not find todo item",
+  })
 
-  return { data }
+  return result
 })
-.decorate('deleteTodoById', (id: number)=> {
-  data = data.filter(item => item.id !== id);
-  return data
 
-})
-.get("/all", ()=> {
-  return { data }
-})
-.get('/todo/:id', ({params: {id}, getTodoById})=>{
-  const response = getTodoById(Number(id))
-  if (!response) {
-    throw new Error("Could not find todo item")
-  }
-  return response
-})
-.post("/todo", ({ body, addNewTodoItem }) => {
-  const { title, description } = body
-  return addNewTodoItem(title, description)
+.post("/todo", ({ body, dbAction, database }) => {
+  const result = dbAction({
+    database: database(),
+    queryMethod: 'run',
+    query: "INSERT INTO todos (title, description, isComplete) VALUES (?, ?, ?)",
+    params: [body.title, body.description, false],
+    errorMessage: "Could add find todo item",
+  })
+
+  return result
 }, {
   body: t.Object({
     title: t.String(),
     description: t.String(),
   })
 })
-.put('/todo/:id', ({body, params: {id}, editTodoById}) => {
-  return editTodoById(Number(id), {...body})
-}, {
-  body: t.Object({
-    title: t.Optional(t.String()),
-    description: t.Optional(t.String()),
-    isComplete: t.Optional(t.Boolean()),
+
+// Not sure what's the best method to update a todo item
+// .put('/todo/:id', ({body, params: {id}, dbAction, database}) => {
+//   const result = dbAction({
+//     database: database(),
+//     queryMethod: 'run',
+//     query: "UPDATE todos SET title = ?, description = ?, isComplete = ? WHERE id = ?",
+//     params: [body.title, body.description, body.isComplete, id],
+//     errorMessage: "Could not update todo item",
+//   })
+// }, {
+//   body: t.Object({
+//     title: t.Optional(t.String()),
+//     description: t.Optional(t.String()),
+//     isComplete: t.Optional(t.Boolean()),
+//   })
+// })
```

Comment:
Will try this out in a different PR,  a bit rusty with backend methodologies. Will research how other people update items